src/vs/platform/native/electron-main/nativeHostMainService.ts

```diff
@@ -212,6 +212,18 @@ export class NativeHostMainService extends Disposable implements INativeHostMain
 		}
 	}
 
+	async updateTitleBarOverlay(windowId: number | undefined, backgroundColor: string, foregroundColor: string): Promise<void> {
+		const window = this.windowById(windowId);
+		if (window?.win) {
+			try {
+				window.win.setTitleBarOverlay({
+					color: backgroundColor,
+					symbolColor: foregroundColor
+				});
+			} catch { }
```

Comment:
Not clear to me why this needs a try/catch and if so why we don't log this error?

Reply:
When I was testing I was switching between custom and native window controls to ensure linux was good but I was still on Windows. This API will throw if there are no native controls to update, but I've added the proper checks now.